src/apiRequest/ApiRequest.jsx

```diff
@@ -0,0 +1,130 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {Row, Col} from 'react-flexbox-grid';
+
+import UrlInput from './components/UrlInput';
+import HttpMethodInput from './components/HttpMethodInput';
+import BodyData from './components/BodyData';
+import ApiRequestForm from './components/ApiRequestForm';
+import SuccessToaster from '../components/SuccessToaster';
+import Button from '../components/Button';
+
+import {injectReducer} from '../app/reducers';
+import apiRequestReducer from './ApiRequestReducer';
+import {getTokenId} from '../auth/AuthSelectors';
+import {getApiResponse} from './ApiRequestSelector';
+import {
+  apiRequest,
+  clearApiResponse,
+} from './ApiRequestActions';
+
+export class ApiRequest extends Component {
+  static onEnter(store) {
+    injectReducer(store, {key: 'apiRequestReducer', apiRequestReducer});
+  }
+
+  static defaultProps = {
+    httpMethods: ['GET', 'POST', 'PUT'],
+    apiResponse: undefined,
+    queryList: [],
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      body: '',
+      method: 'GET',
+      query: '',
+    };
+  }
+
+  handleQueryChange = event => this.setState({query: event.target.value})
+
+  handleMethodChange = event => this.setState({method: event.target.value});
+
+  handleBodyChange = code => this.setState({body: code});
+
+  handleSubmit = () => {
+    const {
+      method,
+      body,
+      query,
+    } = this.state;
+
+    const {
+      baseUrl,
+      authToken,
+      makeApiRequest,
+    } = this.props;
+
+    return () => makeApiRequest(method, `${baseUrl}${query}`, body, authToken);
```

Comment:
You can just call `makeApiRequest()`, so in L96 there will be only `onClick={this.handleSubmit}` :slightly_smiling_face: 
It's better to avoid extra function calls in handlers.